src/plugins/process/README.md

```diff
@@ -0,0 +1,53 @@
+- infos = Information about the process plugin is in keys below
+- infos/author = e1528532 <e1528532@libelektra.org>
+- infos/licence = BSD
+- infos/needs = dump
+- infos/provides =
+- infos/recommends =
+- infos/placements = prerollback rollback postrollback getresolver pregetstorage getstorage postgetstorage setresolver presetstorage setstorage precommit commit postcommit
+- infos/status = maintained unittest shelltests tested configurable experimental memleak
+- infos/metadata =
+- infos/description = executes other plugins in a separate process
+
+## Introduction
+
+The process plugin uses the `pluginprocess` library in order to execute other plugins inside an own process. 
+This is useful for plugins which cause memory leaks to be isolated in an own process. Furthermore 
+this is useful for runtimes or libraries that cannot be reinitialized in the same process after they 
+have been used.
+
+## Usage
+
+Mount this plugin and specify the plugin configuration parameter `plugin` to a plugin that exists on 
+the system. The proxied plugin will then be executed in a separate process and this plugin handles 
+the communication between the processes.
+
+## Dependencies
+
+The `pluginprocess` library, which is only available on POSIX environments currently.
+
+## Examples
+
+```sh
+# Mount the dump plugin a separate process via the process plugin to the cascading namespace `/examples/process`
+sudo kdb mount config.file /examples/process process plugin=dump
```

Comment:
Does not work for me, I get the error:
```
The command bin/kdb mount terminated unsuccessfully with the info:
Was not able to load such a plugin!

Maybe you misspelled it, there is no such plugin or the loader has problems.
You might want to try to set LD_LIBRARY_PATH, use kdb-full or kdb-static.
Errors/Warnings during loading were:

Please report the issue at https://issues.libelektra.org/
```

Even "kdb check process" fails.


Please also add this shell recorder test in tests/shell/shell_recorder/tutorial_wrapper/CMakeLists.txt then hopefully the build server will tell you the same problem I have locally.

Reply:
it is expected that kdb check process fails, as without a plugin configuration it does not work because it doesn't know what plugin shall be proxied.

Reply:
@markus2330 regarding the mounting issue, thats strange, i am pretty sure this worked at some point but i can reproduce it also. Again, logging reveals the actual cause (I've added a simple logging statement for debugging just after it gets the plugin configuration like
```
KeySet * processConfig = elektraPluginGetConfig (handle);
Key * pluginName = ksLookupByName (processConfig, "/plugin", KDB_O_NONE);
ELEKTRA_LOG ("we are trying to open the plugin %s", keyString (pluginName));
```
)

but it looks like the configuration parameter is not present? Other plugins using config seem to handle it the same way though, so i'm not exactly sure yet where the error hides.

```
sudo kdb mount test.dump user/examples/process process plugin=dump
...
src/plugins/process/process.c:62:elektraProcessOpen: we are trying to open the plugin (null)
```

Reply:
You should definitely always add warnings in such situations. It should never be required to enable logging to debug wrong plugin parameters.

> which would have complained about the missing name?

Where do you implement this check? Do you also pass the checkConfig calls to the proxied plugins?

Reply:
> You should definitely always add warnings in such situations.

done

> Where do you implement this check?

Currently it was only in the checkConfig function but now i've also included the same check in the startup function so users get the warning in any case. the little overhead should be acceptable for that.

I don't pass checkConfig calls to proxied plugins though, as there is no Plugin handle in the checkConfig function signature so i can't really access the child process in there.